client/arg.cpp

```diff
@@ -55,6 +55,44 @@ inline int str_startswith(const char *head, const char *worm)
     return !strncmp(head, worm, strlen(head));
 }
 
+/**
+ * Return a pointer to the basename of the file (everything after the
+ * last slash.)  If there is no slash, return the whole filename,
+ * which is presumably in the current directory.
+ **/
+static const char *find_basename(const char *sfile)
+{
+    const char *slash;
+
+    if (!sfile)
+        return sfile;
+
+    slash = strrchr(sfile, '/');
+
+    if (slash == NULL || slash[1] == '\0')
+        return sfile;
+
+    return slash + 1;
+}
+
+/* Some files should always be built locally... */
+static bool
+source_needs_local(const string &filename)
+{
+    const char *p;
+
+    p = find_basename(filename.c_str());
+
+    if (str_startswith("conftest.", p) || str_startswith("tmp.conftest.", p)) {
+#if CLIENT_DEBUG
+        log_info() << "autoconf tests are run locally: " << p << endl;
+#endif
```

Comment:
The message should be shown normally (if debug is set to the info level), the #if shouldn't be there.
Also, the right place for this message should be where the local build is forced (i.e. the "always_local = true" in the other change), as that is the place where the action takes place. If this function got called from multiple place for whatever reason, it'd print the message multiple times.